lua/auto-session.lua

```diff
@@ -89,6 +90,20 @@ local function is_auto_create_enabled()
   return true
 end
 
+-- get the current git branch name, if any, and only if configured to do so
+local function get_branch_name()
+  if AutoSession.conf.auto_session_use_git_branch then
+    local out = vim.fn.systemlist('git rev-parse --abbrev-ref HEAD')
```

Comment:
Just a drive by comment but this function `systemlist` is synchronous and will block nvim till it completes so if this is run at startup it will slow users startups down. An alternative is to use `jobstart` and pass a callback to receive the result, but then the whole outer function will have to adapt to use callbacks as well

Reply:
Yes, I'm aware. Redoing / adapting auto-session to use jobs isn't going to happen. At least not by me. Sorry.

Reply:
@WhoIsSethDaniel no worries, fully a drive by comment, just pointing it out in case you weren't aware

Reply:
may be an idea for #116.  I wonder if the rewrite would benefit from using [plenary](https://github.com/nvim-lua/plenary.nvim). It has excellent job support.

Reply:
Good input, this is something I've thought about a bit, I'm just not sure if the core restoring and saving a session behaviour could actually be offloaded to a job. 🤔 As far as this git functionality, it's a good use-case for a job I'd say. 

I also agree @WhoIsSethDaniel that it's a feature better suited for the rewrite. I'll keep this in mind when working on it.

Thanks for chiming in @akinsho !